### ISSUE TITLE
Edelkrone control

### ISSUE THREAD

**Issue by Operon95** [bitfocus/companion-module-requests#1563]
Any chance of supporting Edelkrone brand of camera movement control devices? If you are familiar with the Edlekrone products of camera movement control devices then you will see an obvious need for Companion integration. (www.edelkrone.com) Whilst Edelkrone has an iOS interface for device configuration and implication, the ability to trigger camera movements from a Stream Deck interface is extant. Furthermore, there exists creating an enhanced integration with BlackMagic ATEM devices.

**Comment by dnmeid** [comment]
The Edelkrone hardware is controlled by Bluetooth and there is no API or possibility for Companion to control it. The Edelkrone APP has no API or protocol that would make it controllable from any external device.

**Comment by philipcain** [comment]
We have an Edelkrone Link Adapter that specifically does provide a documented API. It connects to the devices via Bluetooth or wired connection, and connects to a host computer via USB.

We're doing all sorts of Companion-based control of our Edelkrone hardware using the generic http module.

**Comment by philipcain** [comment]
Specifically, we do the following all via Companion (this would be much better with a module and not custom http commands):

- calibrate button to link and calibrate our head and slide rail
- left/right/up/down/slide left/slide right controls
- store pose
- recall pose
- recall a looping animation between two poses

**Comment by dnmeid** [comment]
Then please attach the API or a permanent link to it here or the chances that somebody picks up this issue are quite low.

**Comment by Operon95** [comment]
Seems like the Edelkrone's Link Adapter has been discontinued. It's not clear that this reflects that they have updated versions of some of their hardware and that a newer Link Adapter will be manufactured to accommodate the newer hardware or they have dispensed with any notions of remote control of their hardware other than what an iOS app can deliver or their handheld remote device which is limited in terms of the number of devices it can control at any one time..

**Comment by 56kbnxc8hw-sketch** [comment]
I agree. It's a huge limitation of Edelkrone's system that it doesn't allow for Bitfocus Companion integration. I have three Edelkrone motion control systems, and it's a nightmare to use them, since even if I leave them setup in my studio, I have to reprogram them every time they power down (to affiliate multiple devices back into a system -- head, focus module, jib, slider, etc.) and to create new key poses (since the devices forget everything when they're powered down). Worst of all, when controlling multiple Edelkrone systems, you need either separate Edelkrone controllers, which themselves are well built but kludgy to use, or separate control surfaces (like different iPhones or iPads for each system). It's a nightmare to use Edelkrone devices in a studio-based creator environment, so my investments in them so far have been a massive waste. They're so close -- as their devices hold so much promise and work so well if you're willing to invest the setup time --  but they're currently optimizing around project-based work. Allowing a Companion module that could remember key pose coordinates and remember which devices are setup to work together would make Edelkrone a dream motion control solution for any studio based creator.

**Comment by josephdadams** [comment]
Closing as **not planned**.

This request is labelled as having no external control protocol (or as unlikely to be feasible) — the device or software doesn't expose an interface Companion could talk to, so a module can't be built for it as things stand.

If the manufacturer adds a control API later, please comment and we'll happily reopen.
